java/client/src/main/java/glide/api/commands/BaseCommands.java

```diff
@@ -1,31 +1,20 @@
 package glide.api.commands;
 
-import glide.ffi.resolvers.RedisValueResolver;
-import glide.managers.BaseCommandResponseResolver;
 import java.util.concurrent.CompletableFuture;
-import response.ResponseOuterClass.Response;
 
-/** Base Commands interface to handle generic command and transaction requests. */
-public interface BaseCommands {
+/**
+ * Base Commands interface to handle generic command and transaction requests.
+ *
+ * @param <T> The data return type.
+ */
+public interface BaseCommands<T> {
 
     /**
-     * Extracts the response from the Protobuf response and either throws an exception or returns the
-     * appropriate response has an Object
+     * Executes a single custom command, without checking inputs. Every part of the command, including
+     * subcommands, should be added as a separate value in args.
      *
-     * @param response Redis protobuf message
-     * @return Response Object
+     * @param args command and arguments for the custom command call
+     * @return CompletableFuture with the response
      */
-    static Object handleObjectResponse(Response response) {
-        // return function to convert protobuf.Response into the response object by
-        // calling valueFromPointer
-        return (new BaseCommandResponseResolver(RedisValueResolver::valueFromPointer)).apply(response);
-    }
-
-    /**
-     * Execute a @see{Command} by sending command via socket manager
-     *
-     * @param args arguments for the custom command
-     * @return a CompletableFuture with response result from Redis
-     */
-    CompletableFuture<Object> customCommand(String[] args);
+    CompletableFuture<T> customCommand(String[] args);
```

Comment:
The whole point of customCommand is that we won't know the return type.  We should return Object in all cases.

Reply:
`T` = `Object` for standalone client
`T` = `RedisValue` for cluster client

Reply:
It would be better to duplicate. Why don't we just implement BaseCommands for the RedisClient, and ClusterBaseCommands for the RedisClusterClient?  I don't see the value to using a generic here when there are only two possible options.

Reply:
Actually, more than two, becase now `T` is `RedisValue<V>` for cluster client. So having `T` we avoid code duplicating N times for sake of maintanability and so on.